webapp/src/Service/ImportProblemService.php

```diff
@@ -277,6 +277,35 @@ public function importZippedProblem(
                         $yamlProblemProperties['name'] = $yamlData['name'];
                     }
                 }
+
+                if (isset($yamlData['type'])) {
+                    $types = explode(' ', $yamlData['type']);
+                    foreach ($types as $type) {
+                        $allowedProblemTypes = ['pass-fail', 'multi-pass', 'scoring', 'interactive', 'submit-answer'];
+                        if (!in_array($type, $allowedProblemTypes)) {
+                           $messages['danger'][] = "Invalid problem type: '$type', must be one of " . implode(', ', $allowedProblemTypes);
+                           return null;
+                        }
+                    }
+                    if (in_array('pass-fail', $types) && in_array('scoring', $types)) {
+                        $messages['danger'][] = "Invalid problem type: 'pass-fail' and 'scoring' are mutually exclusive.";
+                        return null;
+                    }
+                    if (in_array('submit-answer', $types)) {
+                        if (in_array('multi-pass', $types)) {
+                            $messages['danger'][] = "Invalid problem type: 'submit-answer' and 'multi-pass' are mutually exclusive.";
+                            return null;
+                        }
+                        if (in_array('interactive', $types)) {
+                            $messages['danger'][] = "Invalid problem type: 'submit-answer' and 'interactive' are mutually exclusive.";
+                            return null;
+                        }
```

Comment:
I think it would be better to move this validation into the problem class itself: we should never have an inconsistent problem types bitfield.

Reply:
done